assets/cubyz/sbb/tree/oak/1/base_3.zig.zon

```diff
@@ -0,0 +1,22 @@
+.{
+	.blueprint = "cubyz:tree/oak/1/base_3",
+	.children = .{
+		.crimson = .{
+			.{.structure = "cubyz:tree/oak/1/root_1", .chance = 0.125},
+			.{.structure = "cubyz:tree/oak/1/root_2", .chance = 0.125},
+			.{.structure = "cubyz:tree/oak/1/root_3", .chance = 0.125},
+			.{.structure = "cubyz:tree/oak/1/root_4", .chance = 0.125},
+			.{.structure = "cubyz:tree/oak/1/root_5", .chance = 0.125},
+			.{.structure = "cubyz:tree/oak/1/root_6", .chance = 0.125},
+			.{.structure = null, .chance = 0.25},
+		},
+		.brown = .{
+			.{.structure = "cubyz:tree/oak/1/trunk_1"},
+			.{.structure = "cubyz:tree/oak/1/trunk_2"},
+			.{.structure = "cubyz:tree/oak/1/trunk_3"},
+			.{.structure = "cubyz:tree/oak/1/trunk_4"},
+			.{.structure = "cubyz:tree/oak/1/trunk_5"},
+			.{.structure = "cubyz:tree/oak/1/trunk_6"},
```

Comment:
@Argmaster should we add some kind of wildcard system?
Then here it would just be a single entry:
```zig.zon
.{.structure = "cubyz:tree/oak/1/trunk_*"},
```

Reply:
Alright, but what are the rules?
The obvious matches:

```
cubyz:tree/oak/1/trunk_0
cubyz:tree/oak/1/trunk_1
cubyz:tree/oak/1/trunk_453
```

But, does that match to?

```
cubyz:tree/oak/1/trunk_/0
cubyz:tree/oak/1/trunk_/1
cubyz:tree/oak/1/trunk_/234
```

Do we go full glob with `?` and `**`?
What do we do with `cubyz:tree/oak/1/trunk_*/foobar_1`, do we even allow this?
Where is the wiki explaining what is and what is not allowed C:
Maybe instead of wildcards we would allow specifying directory-like syntax and we would take *everything starting with `cubyz:tree/oak/1/trunk/`*

Reply:
The last one creates grouping that is visible both in code **and in directory structure** with no tricks to be done.

Reply:
Generally speaking I am not happy with the `generator` structure types, they are clever trick but not indented use, it would be best to solve both those problems at once.

Reply:
> Alright, but what are the rules?

In my opinion it should only match one directory at a time, just like in bash.

> Do we go full glob with ? and **?

Let's stick to the simple things, and extend when we need it.

> What do we do with cubyz:tree/oak/1/trunk_*/foobar_1, do we even allow this?

Sure, that doesn't seem too difficult.

> Maybe instead of wildcards we would allow specifying directory-like syntax and we would take everything starting with cubyz:tree/oak/1/trunk/

That would be another option, I guess. It does however limit your freedom for organizing things.

Reply:
You know what I think about freedom.